policy/modules/system/systemd-homed.te

```diff
@@ -0,0 +1,235 @@
+systemd_unit_file(systemd_homed_unit_file_t)
+
+#######################################
+#
+# systemd_homed local policy
+#
+
+allow systemd_homed_t self:capability { sys_admin sys_resource };
+allow systemd_homed_t self:capability { dac_override dac_read_search setuid setgid };
+allow systemd_homed_t self:netlink_kobject_uevent_socket create_socket_perms;
+allow systemd_homed_t self:unix_dgram_socket create_socket_perms;
+
+domtrans_pattern(systemd_homed_t, systemd_homework_exec_t, systemd_homework_t)
+allow systemd_homed_t systemd_homework_t:process2 nnp_transition;
+
+# homed.conf
+files_read_config_files(systemd_homed_t)
+init_read_pid_files(systemd_homed_t)
+libs_read_lib_files(systemd_homed_t)
+
+# /var/home
+files_watch_home(systemd_homed_t)
+files_search_home(systemd_homed_t)
+
+# labeled home directories
+userdom_home_manager(systemd_homed_t)
+userdom_manage_home_role(system_r, systemd_homed_t)
+
+# unlabeled home directories
+files_manage_isid_type_dirs(systemd_homed_t)
+files_manage_isid_type_files(systemd_homed_t)
+
+# /var/lib/systemd/home
+manage_files_pattern(systemd_homed_t, systemd_homed_library_dir_t, systemd_homed_record_t)
+init_var_lib_filetrans(systemd_homed_t, systemd_homed_library_dir_t, dir, "home")
+filetrans_pattern(systemd_homed_t, systemd_homed_library_dir_t, systemd_homed_record_t, file)
+
+# /var/run/systemd/home
+create_dirs_pattern(systemd_homed_t, init_var_run_t, systemd_homed_runtime_dir_t)
+init_pid_filetrans(systemd_homed_t, systemd_homed_runtime_dir_t, dir, "home")
+
+# /var/run/systemd/home/*.dont-suspend
+manage_fifo_files_pattern(systemd_homed_t, systemd_homed_runtime_dir_t, systemd_homed_runtime_pipe_t)
+filetrans_pattern(systemd_homed_t, systemd_homed_runtime_dir_t, systemd_homed_runtime_pipe_t, fifo_file)
+
+# /var/run/systemd/home/notify
+create_sock_files_pattern(systemd_homed_t, systemd_homed_runtime_dir_t, systemd_homed_runtime_socket_t)
+delete_sock_files_pattern(systemd_homed_t, systemd_homed_runtime_dir_t, systemd_homed_runtime_socket_t)
+filetrans_pattern(systemd_homed_t, systemd_homed_runtime_dir_t, systemd_homed_runtime_socket_t, sock_file, "notify")
+
+# /var/run/systemd/userdb/io.systemd.Home
+systemd_manage_userdbd_runtime_sock_files(systemd_homed_t)
+add_entry_dirs_pattern(systemd_homed_t, systemd_userdbd_runtime_t, systemd_userdbd_runtime_t)
+filetrans_pattern(systemd_homed_t, systemd_userdbd_runtime_t, systemd_userdbd_runtime_t, sock_file, "io.systemd.Home")
+
+# /tmp
+rw_files_pattern(systemd_homed_t, systemd_homed_tmpfs_t, systemd_homed_tmpfs_t)
+fs_tmpfs_filetrans(systemd_homed_t, systemd_homed_tmpfs_t, file)
+
+kernel_dgram_send(systemd_homed_t)
+kernel_read_system_state(systemd_homed_t)
+
+auth_use_nsswitch(systemd_homed_t)
+
+dev_getattr_generic_blk_files(systemd_homed_t)
+dev_read_sysfs(systemd_homed_t)
+
+fs_getattr_cgroup(systemd_homed_t)
+fs_getattr_xattr_fs(systemd_homed_t)
+fs_search_cgroup_dirs(systemd_homed_t)
+fs_write_cgroup_files(systemd_homed_t)
```

Comment:
which cgroup files need to be written to?

Reply:
type=AVC msg=audit(1725706986.512:619):  avc:  denied  { write } for  pid=903 comm="systemd-homed" name="memory.pressure" dev="cgroup2" ino=4616 scontext=system_u:system_r:systemd_homed_t:s0 tcontext=system_u:object_r:cgroup_t:s0 tclass=file permissive=1

Reply:
This is unfortunate, but currently there is no other way, either allow write to all cgroup files or to none.